lib/Crypto/DecryptAll.php

```diff
@@ -55,90 +59,149 @@ class DecryptAll {
 	protected $userManager;
 
 	/**
+	 * DecryptAll constructor.
+	 *
 	 * @param Util $util
 	 * @param KeyManager $keyManager
 	 * @param Crypt $crypt
 	 * @param Session $session
+	 * @param IUserManager $userManager
 	 * @param QuestionHelper $questionHelper
+	 * @param EnvironmentHelper $environmentHelper
 	 */
 	public function __construct(
 		Util $util,
 		KeyManager $keyManager,
 		Crypt $crypt,
 		Session $session,
 		IUserManager $userManager,
-		QuestionHelper $questionHelper
+		QuestionHelper $questionHelper,
+		EnvironmentHelper $environmentHelper
 	) {
 		$this->util = $util;
 		$this->keyManager = $keyManager;
 		$this->crypt = $crypt;
 		$this->session = $session;
 		$this->userManager = $userManager;
 		$this->questionHelper = $questionHelper;
+		$this->environmentHelper = $environmentHelper;
 	}
 
 	/**
 	 * prepare encryption module to decrypt all files
 	 *
+	 * - The value of $user (when called from decrypt-all command ) is as follows:
+	 *   the uid of user passed, if the oC filesystem is encrypted using user-keys
+	 *   the empty string '', if the oC filesystem is encrypted using master-key
+	 *
 	 * - Throws LoginException when user login fails either recovery password fails
 	 *   or if the user password fails
 	 *
 	 * @param InputInterface $input
 	 * @param OutputInterface $output
-	 * @param $user
+	 * @param string $user
 	 * @return bool
 	 * @throws LoginException
 	 */
 	public function prepare(InputInterface $input, OutputInterface $output, $user) {
-
-		$question = new Question('Please enter the recovery key password: ');
-
+		// We need to setup the module with everything it needs in order to decrypt user data
+		// This could be using master key or user key encryption
+		// and could also be for a single user or multiple users
+
+		/**
+		 * Try to stop decrypting process if password method is used for decrypting all users.
+		 * If the decrypt-all command doesn't provide a user, which means decryption of oC
```

Comment:
this is too long, no one will read this. PHPdoc needs to say that $user can be null and tell in what situation